Update Asset Inventory to match acceptance criteria

albertoblaz opened the pull request:
## Summary

It closes https://github.com/elastic/kibana/issues/217079

- Update implementation as per the the Asset Inventory [Enable Users to Search, Filter, and Group Assets within Asset Inventory acceptance criteria](https://github.com/elastic/security-team/issues/10344):
    - Set the default columns for the data table as it is specified in 
    - Update filter attributes
- Update implementation to match answers given in [this epic](https://github.com/elastic/security-team/issues/11856):
    - Hide 3-dots button on each data table row
    - Remap bar chart fields to `entity.type` and `entity.sub_type`

## Screenshot

<img width="2053" alt="image" src="https://github.com/user-attachments/assets/6bdb50d3-b32b-40b3-ad97-0d5eb70f8f2c" />

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

albertoblaz committed:
Set new default columns for data grid

albertoblaz committed:
Update displayed attributes and mappings in filters

albertoblaz committed:
Hide 3-dots button on each data table row

albertoblaz committed:
Remap barchart fields to type & sub_type

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/289546)
* Commit: 2e0c5496ee1da42bb11cead2f9bfae96e3b0fc2e

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/289546#0195f6ea-3097-400d-904c-bce0a2e6cec4)

cc @albertoblaz

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289546","number":289546,"commit":"2e0c5496ee1da42bb11cead2f9bfae96e3b0fc2e"}],"number":289546}
buildkite-pr-comment-->

albertoblaz committed:
Remove unused translations

opauloh commented on x-pack/solutions/security/plugins/security_solution/public/asset_inventory/components/asset_inventory_data_table.tsx:
```diff
@@ -337,21 +334,22 @@ export const AssetInventoryDataTable = ({
-      ),
-    },
-  ];
+  // TODO Decide if we need to keep, hide or remove the 3-dots button on each row
+  // const externalControlColumns: RowControlColumn[] = [
+  //   {
+  //     id: 'more-actions',
+  //     headerAriaLabel: moreActionsLabel,
+  //     headerCellRender: () => null,
+  //     renderControl: () => (
+  //       <EuiButtonIcon
+  //         aria-label={moreActionsLabel}
+  //         iconType="boxesHorizontal"
+  //         color="primary"
+  //         isLoading={isLoadingGridData}
+  //       />
+  //     ),
+  //   },
+  // ];
```
We might still use this, but maybe instead of commenting the code, we could add a flag to show/hide it for now, or just remove

albertoblaz replied:
I removed the code. If we need to reintroduce it, we can look back at this PR

opauloh commented on x-pack/solutions/security/plugins/security_solution/public/asset_inventory/components/filters/asset_inventory_filters.tsx:
```diff
@@ -24,25 +24,25 @@ const DEFAULT_ASSET_INVENTORY_FILTERS: FilterControlConfig[] = [
       defaultMessage: 'Type',
     }),
-    fieldName: 'entity.category',
+    fieldName: 'entity.type',
```
I think we should have those fields coming from the constants file

opauloh replied:
Also I noticed we are missing a testing file for this component, it would be nice to add test coverage where it's missing to have testings that resembles the acceptance criteria

albertoblaz replied:
💯 I focused on getting the view built but there are no tests for filters, data table, chart or search bar. We need to add those

opauloh commented on x-pack/solutions/security/plugins/security_solution/public/asset_inventory/hooks/use_fetch_chart_data.ts:
```diff
@@ -25,18 +25,18 @@ const getTopAssetsQuery = ({ query, sort }: UseTopAssetsOptions) => ({
+    entityType: {
       terms: {
-        field: 'entity.category',
+        field: 'entity.type',
```
it would be great to have those fields coming from the constants file as well

albertoblaz replied:
Done!

opauloh commented on x-pack/solutions/security/plugins/security_solution/public/asset_inventory/hooks/use_fetch_chart_data.ts:
```diff
@@ -117,30 +117,30 @@ const tooltipOtherLabel = i18n.translate(
-//   { category: 'cloud-compute', source: 'aws-security', count: 300, },
-//   { category: 'cloud-storage', source: 'gcp-compute', count: 221, },
-//   { category: 'cloud-storage', source: 'aws-security', count: 117, },
+//   { type: 'cloud-compute', subType: 'gcp-compute', count: 500, },
+//   { type: 'cloud-compute', subType: 'aws-security', count: 300, },
+//   { type: 'cloud-storage', subType: 'gcp-compute', count: 221, },
+//   { type: 'cloud-storage', subType: 'aws-security', count: 117, },
```
IMO, having the expected output in a test file will ensure the code resembles the acceptance criteria here (checking if it's grouped by `entity.type` and `entity.sub_type`) and ensure the examples are up to date as the tests break if implementation changes, while comments can go unnoticed

albertoblaz replied:
Deleted comment but still missing tests

opauloh replied:
Thanks, I added the missing tests for this hook!

opauloh reviewed (commented):
Thanks for addressing the acceptance criteria, I left some small improvements for you to consider adding before merging the PR but it looks great already

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/289615)
* Commit: 3704e95914ecf81905cefdd7dfe4c21c47e40ed2

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/cc09a96efea1795a62a1fc059fad525e7a10def3) | [after](https://github.com/elastic/kibana/commit/3704e95914ecf81905cefdd7dfe4c21c47e40ed2) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 8.9MB | 8.9MB | -412.0B |



### History
* :broken_heart: [Build #289546](https://buildkite.com/elastic/kibana-pull-request/builds/289546) failed 2e0c5496ee1da42bb11cead2f9bfae96e3b0fc2e

cc @albertoblaz

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289615","number":289615,"commit":"3704e95914ecf81905cefdd7dfe4c21c47e40ed2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289546","number":289546,"commit":"2e0c5496ee1da42bb11cead2f9bfae96e3b0fc2e"}],"number":289615}
buildkite-pr-comment-->

albertoblaz committed:
Remove 3-dots button

albertoblaz committed:
Delete old comment

albertoblaz committed:
Map attribute keys to constants.

- Remap grouping fields category->type and type->source

albertoblaz committed:
Remove comment

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/289794)
* Commit: 6b6f39db3e03cb855ea8f21ee87a50e675513d31

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/289794#0195fadd-7512-4051-853b-9f3dd5e3a313)

### History
* :green_heart: [Build #289615](https://buildkite.com/elastic/kibana-pull-request/builds/289615) succeeded 3704e95914ecf81905cefdd7dfe4c21c47e40ed2
* :broken_heart: [Build #289546](https://buildkite.com/elastic/kibana-pull-request/builds/289546) failed 2e0c5496ee1da42bb11cead2f9bfae96e3b0fc2e

cc @albertoblaz

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289794","number":289794,"commit":"6b6f39db3e03cb855ea8f21ee87a50e675513d31"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289615","number":289615,"commit":"3704e95914ecf81905cefdd7dfe4c21c47e40ed2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289546","number":289546,"commit":"2e0c5496ee1da42bb11cead2f9bfae96e3b0fc2e"}],"number":289794}
buildkite-pr-comment-->

albertoblaz committed:
Map chart attributes to constants

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

opauloh committed:
adding entity id checking

opauloh committed:
Refactors chart data fetching into composable functions

Improves code organization and testability by extracting
the query construction and data transformation logic into
separate, reusable functions. This also allows to add unit tests for them.

opauloh committed:
Disables query retries in test provider

Ensures that queries in the test environment do not automatically retry on failure.
This provides more deterministic and reliable test results by preventing potential masking of underlying issues due to automatic retries.

opauloh committed:
Updates asset inventory chart data flow

Refactors the asset inventory bar chart to consume data from a new data structure.

This change improves data handling and consistency between the chart component and the data fetching logic.

opauloh committed:
Merge branch 'main' into asset-inv-default-cols

opauloh committed:
reverting flyout changes

opauloh committed:
Merge remote-tracking branch 'albertoblaz/asset-inv-default-cols' into asset-inv-default-cols

opauloh committed:
Allows left controls to grow in data grid

Makes the left controls within the data grid expandable improving layout consistency.

opauloh committed:
Allows data grid left controls to grow on cloud security datatable

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/290471)
* Commit: 0f541a5e2244ef0daf0ea8c1302035429bd89306

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/62b418b90712b42f3893052f851c6076c6df9363) | [after](https://github.com/elastic/kibana/commit/0f541a5e2244ef0daf0ea8c1302035429bd89306) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 7182 | 7184 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/62b418b90712b42f3893052f851c6076c6df9363) | [after](https://github.com/elastic/kibana/commit/0f541a5e2244ef0daf0ea8c1302035429bd89306) | diff |
| --- | --- | --- | --- |
| `cloudSecurityPosture` | 513.0KB | 513.1KB | +42.0B |
| `securitySolution` | 8.9MB | 8.9MB | +55.0B |
| total |  |  | +97.0B |



### History
* :broken_heart: [Build #289794](https://buildkite.com/elastic/kibana-pull-request/builds/289794) failed 6b6f39db3e03cb855ea8f21ee87a50e675513d31
* :green_heart: [Build #289615](https://buildkite.com/elastic/kibana-pull-request/builds/289615) succeeded 3704e95914ecf81905cefdd7dfe4c21c47e40ed2
* :broken_heart: [Build #289546](https://buildkite.com/elastic/kibana-pull-request/builds/289546) failed 2e0c5496ee1da42bb11cead2f9bfae96e3b0fc2e

cc @albertoblaz @opauloh

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290471","number":290471,"commit":"0f541a5e2244ef0daf0ea8c1302035429bd89306"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289794","number":289794,"commit":"6b6f39db3e03cb855ea8f21ee87a50e675513d31"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289615","number":289615,"commit":"3704e95914ecf81905cefdd7dfe4c21c47e40ed2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289546","number":289546,"commit":"2e0c5496ee1da42bb11cead2f9bfae96e3b0fc2e"}],"number":290471}
buildkite-pr-comment-->
